lib/api/createreadstream.js

```diff
@@ -0,0 +1,148 @@
+var RS = require('readable-stream')
+  , SMB2Forge = require('../tools/smb2-forge')
+  , SMB2Request = SMB2Forge.request
+  , bigint = require('../tools/bigint')
+  ;
+
+/*
+* createReadStream
+* ================
+*
+* Return a read stream for a file on the share
+*/
+module.exports = function(filename, options){
+  options = processOptions(options);
+
+  var connection = this
+    , file
+    , bufferedChunks = []
+    , chunckPushed = 0
+    , chunckRequested = 0
+    , opened = false
+    , fileLength = 0
+    , offset = new bigint(8).add(options.start)
+    , stop = false
+    , nbRemainingPackets = 0
+    , maxPacketSize = 0x00010000
+    , readable = new RS.Readable({
+        read: function(size) {
+          if (opened) {
+            readNext();
+          }
+        }
+      })
+    ;
+  
+  var close = function () {
+    stop = true;
+    if (opened) {
+      opened = false;
+      SMB2Request('close', file, connection, function(err){
+        if(err){
+          readable.emit('error', err);
+        }
+      });
+    }
+  };
+
+  SMB2Request('open', {path:filename}, connection, function(err, openedFile) { 
+    file = openedFile;
+    opened = true;
+    if(err) {
+      readable.emit('error', err);
+      return close();
+    } else {
+      fileLength = file.EndofFile.readUInt16LE();
```

Comment:
whoops I think it should be `readUInt32LE()` sorry
```suggestion
      fileLength = file.EndofFile.readUInt32LE();
```